File: src/rdebug.cpp
//--------------------------------------------------------------------------//
/// Copyright (c) 2017 by Milos Tosic. All Rights Reserved.                /// 
/// License: http://www.opensource.org/licenses/BSD-2-Clause               ///
//--------------------------------------------------------------------------//

#include <rdebug_pch.h>

#define RTM_LIBHANDLER_DEFINE
#define RBASE_NAMESPACE rdebug
#include <rbase/inc/libhandler.h>

namespace rdebug {

bool init(rtm::LibInterface* _libInterface)
{
	g_allocator		= _libInterface ? _libInterface->m_memory : 0;
	g_errorHandler	= _libInterface ? _libInterface->m_error  : 0;

	return false;
}

void shutDown()
{
}

} // namespace rdebug

File: scripts/rdebug.lua
--
-- Copyright 2025 Milos Tosic. All rights reserved. 
-- License: http://www.opensource.org/licenses/BSD-2-Clause
--

function projectDependencies_rdebug()
	return { "rbase", "DIA" }
end

function projectExtraConfig_rdebug()
	configuration { "gmake" }
		if "mingw-gcc" == _OPTIONS["gcc"] then -- on windows, we patch heap functions, no need to wrap malloc family of funcs
			buildoptions { "-Wno-unknown-pragmas" }
		end
	configuration { "linux-*" }
		buildoptions { "-Wimplicit-fallthrough=0" }
	configuration {}
	includedirs	{ path.join(projectGetPath("rdebug"), "../") }
end

function projectExtraConfigExecutable_rdebug() 
	includedirs	{ path.join(projectGetPath("rdebug"), "../") }
end

function projectAdd_rdebug()
	addProject_lib("rdebug", LibraryType.Tool)
end
